sources/packages/backend/apps/workers/src/services/disbursement-schedule/disbursement-schedule.service.ts

```diff
@@ -0,0 +1,179 @@
+import { Injectable } from "@nestjs/common";
+import {
+  ApplicationStatus,
+  DisbursementSchedule,
+  MSFAANumber,
+  OfferingIntensity,
+  RecordDataModelService,
+} from "@sims/sims-db";
+import { CustomNamedError } from "@sims/utilities";
+
+import { DataSource, IsNull, Not } from "typeorm";
+import { MSFAANumberService } from "..";
+import {
+  APPLICATION_MSFAA_ALREADY_ASSOCIATED,
+  DISBURSEMENT_NOT_FOUND,
+} from "../../constants";
+
+@Injectable()
+export class DisbursementScheduleService extends RecordDataModelService<DisbursementSchedule> {
+  constructor(
+    dataSource: DataSource,
+    private readonly msfaaNumberService: MSFAANumberService,
+  ) {
+    super(dataSource.getRepository(DisbursementSchedule));
+  }
+
+  /**
+   * Associates an MSFAA number to the disbursement(s) checking
+   * whatever is needed to create a new MSFAA or use an
+   * existing one instead.
+   * @param assessmentId assessment id of the workflow instance.
+   */
+  async associateMSFAANumber(assessmentId: number): Promise<void> {
+    const [firstDisbursement, secondDisbursement] = await this.repo.find({
+      select: {
+        id: true,
+        studentAssessment: {
+          id: true,
+          application: { id: true, student: { id: true } },
+          offering: { id: true, offeringIntensity: true, studyStartDate: true },
+        },
+        msfaaNumber: {
+          id: true,
+        },
+      },
+      relations: {
+        studentAssessment: { application: { student: true }, offering: true },
+        msfaaNumber: true,
+      },
+      where: {
+        studentAssessment: { id: assessmentId },
+      },
+    });
+
+    if (!firstDisbursement) {
+      throw new CustomNamedError(
+        "Disbursement not found or one of its associations is missing.",
```

Comment:
what do you mean by `one of its associations is missing` ??  I think only `Disbursement not found` is needed right? am I missing anything

Reply:
I would be ok with the short message as @ann-aot mentioned also.